python/ray/serve/_private/deployment_state.py

```diff
@@ -2317,8 +2323,13 @@ def order(deadline: int):
 
         return to_stop, remaining
 
-    def _migrate_replicas_on_draining_nodes(self):
+    def _migrate_replicas_on_draining_nodes(self, in_steady_state: bool):
         draining_node_deadlines = self._cluster_node_info_cache.get_draining_nodes()
+        if RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY and in_steady_state:
+            node, deadline = self._deployment_scheduler.detect_compact_opportunities()
+            if node:
+                draining_node_deadlines[node] = deadline
```

Comment:
hmm I don't really like how deeply nested this dependency on the deployment scheduler is... how about passing in the `draining_node_deadlines` to this function and we can decide to modify that list farther up the call stack? that would also pull up the `self._cluster_node_info_cache` dependency which I see as a win.

So the call signature for this function would be
`def _migrate_replicas_on_draining_nodes(self, draining_nodes: Dict[node, deadline]):`